### PR TITLE
feat/cli

### PR DESCRIPTION
**Pull request by brian-mulier-p**
closes kestra-io/kestra#1429

### CODE REVIEW

**Comment by brian-mulier-p** [comment]
Some uncertainties:

- No way to provide the --identity authentication as it is the same issue as with GCP: since we're running inside docker, even though running a VM inside Azure provide a managed-identity to the VM, it does not transit to the created Docker container. I feel we have to implement the PROCESS runner on those soon enough to allow such authentication.

- Couldn't find another way to properly test the task. Even though I created an Azure account and was able to test it on my side, I don't have the credentials under the hand and can't access the github secrets to update the Azure account & Kestra test configuration to include such authentication. I went for the unit test on a package-scoped method here (although I didn't test that the tested method was called, I feel bad but idk).

Tell me if you have some ideas :+1: 

Also, I have the problem regarding tests here since Azure does not provide any way to authenticate through environment variables (which are test-friendly) while other cloud providers do.

File: src/main/java/io/kestra/plugin/azure/cli/AzCLI.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            title = "Additional environment variables for the current process."
+    )
+    @PluginProperty(
+            additionalProperties = String.class,
+            dynamic = true
+    )
+    protected Map<String, String> env;
+
+    @Schema(
+            title = "Docker options when for the `DOCKER` runner."
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/azure/cli/AzCLI.java:107]:
```suggestion
            title = "Docker options for the `DOCKER` runner."
```

File: src/main/java/io/kestra/plugin/azure/cli/AzCLI.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    @Schema(
+            title = "Tenant id to use."
+    )
+    @PluginProperty(dynamic = true)
+    private String tenant;
+
+    @Schema(
+            title = "Is the account a service principal ?"
+    )
+    @PluginProperty(dynamic = true)
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/azure/cli/AzCLI.java:94]: Usually boolean are not dynamic properties

File: src/main/java/io/kestra/plugin/azure/cli/AzCLI.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        List<String> loginCommands = new ArrayList<>();
+        if (this.username != null) {
+            StringBuilder loginCommand = new StringBuilder("az login -u ").append(this.username);
+
+            if (this.password != null) {
+                loginCommand.append(" -p ").append(this.password);
+            }
+            if (this.tenant != null) {
+                loginCommand.append(" --tenant ").append(this.tenant);
+            }
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/azure/cli/AzCLI.java:145]: username, password and tenant should be rendered

File: src/main/java/io/kestra/plugin/azure/cli/AzCLI.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    @Schema(
+            title = "The commands to run."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    @NotEmpty
+    private List<String> commands;
+
+    @Schema(
+            title = "Account username."
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/azure/cli/AzCLI.java:74]: Explain here or in the task description that if a username is set, the task will emit an `as login` command prior to the commands.

File: src/main/java/io/kestra/plugin/azure/cli/AzCLI.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+                .withRunnerType(RunnerType.DOCKER)
+                .withDockerOptions(this.docker)
+                .withCommands(
+                        ScriptService.scriptCommands(
+                                List.of("/bin/sh", "-c"),
+                                loginCommands,
+                                this.commands)
+                );
+
+        commands = commands.withEnv(this.env);
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/azure/cli/AzCLI.java:130]: It is not rendered whereas the property is dynamic

**Review by loicmathieu — CHANGES_REQUESTED** [review]

File: src/main/java/io/kestra/plugin/azure/cli/AzCLI.java (hunk 5 above)
Comment (brian-mulier-p) [reply src/main/java/io/kestra/plugin/azure/cli/AzCLI.java:130]: The env are already rendered inside the CommandsWrapper

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**